offload_vio/server_rx/plugin.cpp

```diff
@@ -0,0 +1,78 @@
+#include "common/plugin.hpp"
+#include "common/switchboard.hpp"
+#include "common/data_format.hpp"
+#include "common/phonebook.hpp"
+
+#include <ecal/ecal.h>
+#include <ecal/msg/protobuf/subscriber.h>
+
+#include "vio_input.pb.h"
+
+using namespace ILLIXR;
+
+class server_reader : public plugin {
+public:
+	server_reader(std::string name_, phonebook* pb_)
+		: plugin{name_, pb_}
+		, sb{pb->lookup_impl<switchboard>()}
+		, _m_imu_cam{sb->get_writer<imu_cam_type>("imu_cam")}
+    { 
+		eCAL::Initialize(0, NULL, "VIO Offloading Sensor Data Reader");
+		subscriber = eCAL::protobuf::CSubscriber
+		<vio_input_proto::IMUCamVec>("vio_input");
+		subscriber.AddReceiveCallback(
+		std::bind(&server_reader::ReceiveVioInput, this, std::placeholders::_2));
+	}
+
+	virtual void _p_one_iteration() {
+	}
+
+
+private:
+	void ReceiveVioInput(const vio_input_proto::IMUCamVec& vio_input) {	
+		
+		// Loop through all IMU values first then the cam frame	
+		for (int i = 0; i < vio_input.imu_cam_data_size(); i++) {
+			vio_input_proto::IMUCamData curr_data = vio_input.imu_cam_data(i);
+
+			std::optional<cv::Mat> cam0 = std::nullopt;
+			std::optional<cv::Mat> cam1 = std::nullopt;
+
+			if (curr_data.rows() != -1 && curr_data.cols() != -1) {
+				size_t img0_size = curr_data.rows() * curr_data.cols();
+				char *img0_buffer = new char[img0_size + 1];
+				memcpy(img0_buffer, curr_data.img0_data().c_str(), img0_size + 1);
```

Comment:
We should avoid dealing with raw pointers and `memcpy` if possible. Will [this][1] constructor work?
```
<template typename _Tp>
cv::Mat(
    const std::vector< _Tp > &vec,
    bool copyData=false,
)
```

[1]: https://docs.opencv.org/3.4/d3/d63/classcv_1_1Mat.html#a23b182c4ffd46abe38f460c1480ae887

Reply:
It did not, but doing a deep copy of the data before creating the cv::Mat did